src/vita-make-fself.c

```diff
@@ -171,23 +178,34 @@ int main(int argc, const char **argv) {
 	ver.unk3 = 16;
 	ver.unk4 = 0;
 
-	SCE_controlinfo_5 control_5 = { 0 };
-	control_5.common.type = 5;
-	control_5.common.size = sizeof(control_5);
-	control_5.common.unk = 1;
-	SCE_controlinfo_6 control_6 = { 0 };
-	control_6.common.type = 6;
-	control_6.common.size = sizeof(control_6);
-	control_6.common.unk = 1;
-	control_6.is_used = 1;
-	if (mem_budget) {
-		control_6.attr = attribute_cinfo;
-		control_6.phycont_memsize = phycont_mem_budget;
-		control_6.total_memsize = mem_budget;	
+
+	SCE_elf_digest_info control_info_digest = { 0 };
+	control_info_digest.head.type = SCE_ELF_DIGEST_INFO;
+	control_info_digest.head.size = sizeof(SCE_elf_digest_info);
+	control_info_digest.head.has_next = true;
+	memcpy(control_info_digest.constant, digest_constant, sizeof(digest_constant));
+	if(sha256_file(input_path, control_info_digest.elf_digest) != 0) {
+		perror("Cannot generate ELF digest");
+		goto error;
 	}
-	SCE_controlinfo_7 control_7 = { 0 };
-	control_7.common.type = 7;
-	control_7.common.size = sizeof(control_7);
+	control_info_digest.min_required_fw = 0x360;
```

Comment:
Also, since the sce_sdk_version of vitasdk is 3.57, it will be 0x357 instead of 0x360.

Reply:
I actually looked through all system self files, all of them have that field set to 0 so I think It should actually just be kept at 0 like before

Reply:
well, yeah. can set 0 also.